src/core/thread/mle.cpp

```diff
@@ -1560,6 +1560,7 @@ ThreadError Mle::SendChildUpdateRequest(void)
     VerifyOrExit((message = NewMleMessage()) != NULL, ;);
     SuccessOrExit(error = AppendHeader(*message, Header::kCommandChildUpdateRequest));
     SuccessOrExit(error = AppendMode(*message, mDeviceMode));
+    SuccessOrExit(error = AppendSourceAddress(*message));
```

Comment:
Thread spec is not clear in this case. I've found it in 4.5.1 and 4.12.3 but I'm now sure if it applies to this case.
Certification harness checks if Source Address TLV is present in Child Update sent by a detached child. That's why I added it but I'm not sure if harness is compliant with the spec.